dbms/src/Storages/DeltaMerge/DMDecoratorStreams.h

```diff
@@ -24,14 +26,35 @@ namespace DB
 {
 namespace DM
 {
+
+static constexpr size_t UNROLL_BATCH = 64;
+
+/// DMColumnFilterBlockInputStream is used to filter the column and filter out the rows whose del_mark is true
```

Comment:
I think it will be better to use a new class to do the "del_mark filtering" job.